src/process_utility.c

```diff
@@ -269,6 +269,46 @@ reindex_chunk(Oid chunk_relid, void *arg)
 	}
 }
 
+static bool
+process_drop(Node *parsetree)
+{
+	DropStmt *stmt = (DropStmt *) parsetree;
+	ListCell   *cell1;
+	Cache	   *hcache = NULL;
+	bool handled = false;
+
+	if(stmt->removeType != OBJECT_TABLE)
+	{
+		return false;
+	}
+	
+	hcache = hypertable_cache_pin();
+
+	foreach(cell1, stmt->objects)
+	{
+		List	   *object = lfirst(cell1);
+		Oid			relid = RangeVarGetRelid(makeRangeVarFromNameList(object), NoLock, true);
+		if (OidIsValid(relid)){
+			Hypertable *ht;
+
+			ht = hypertable_cache_get_entry(hcache, relid);
+			if (NULL != ht)
+			{
+				if (list_length(stmt->objects) != 1)
+				{
+					elog(ERROR, "Cannot drop a hypertable along with other objects");
+				}
+				OidFunctionCall1(catalog_get_internal_function_id(catalog_get(), DDL_DROP_HYPERTABLE),
```

Comment:
can we simplify this? Wrap in a function or macro? three nested functions in one line seems a bit messy.

Reply:
fixed